timelock-impl/src/test/java/com/palantir/atlasdb/timelock/management/DiskNamespaceLoaderTest.java

```diff
@@ -17,38 +17,73 @@
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.mockito.Mock;
 
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.TimelockNamespaces;
 import com.palantir.atlasdb.timelock.paxos.PaxosTimeLockConstants;
-import com.palantir.paxos.Client;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.paxos.SqliteConnections;
+import com.palantir.tokens.auth.AuthHeader;
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
 
 public class DiskNamespaceLoaderTest {
     private static final String NAMESPACE_1 = "namespace_1";
     private static final String NAMESPACE_2 = "namespace_2";
+    private static final AuthHeader AUTH_HEADER = AuthHeader.valueOf("Bearer omitted");
+    @Mock private Function<String, TimeLockServices> serviceFactory;
+    @Mock private Supplier<Integer> maxNumberOfClientsSupplier;
+
+    private final MetricsManager metricsManager = new MetricsManager(
+            new MetricRegistry(),
+            DefaultTaggedMetricRegistry.getDefault(),
+            unused -> false);
+    private TimeLockManagementResource timeLockManagementResource;
 
     @Rule
     public final TemporaryFolder tempFolder = new TemporaryFolder();
 
-    public DiskNamespaceLoader diskNamespaceLoader;
-
     @Before
-    public void setup() {
-        diskNamespaceLoader = new DiskNamespaceLoader(tempFolder.getRoot().toPath());
+    public void setup() throws MalformedURLException {
+        URL testUrl = new URL("http", "host", "file");
+        RedirectRetryTargeter redirectRetryTargeter = RedirectRetryTargeter
+                .create(testUrl, ImmutableList.of(testUrl));
+
+        Path rootFolderPath = tempFolder.getRoot().toPath();
+        PersistentNamespaceContext persistentNamespaceContext = PersistentNamespaceContext.of(rootFolderPath, SqliteConnections
+                        .createDefaultNamedSqliteDatabaseAtPath(rootFolderPath));
+
+        TimelockNamespaces namespaces = new TimelockNamespaces(metricsManager, serviceFactory, maxNumberOfClientsSupplier);
+
+        timeLockManagementResource = TimeLockManagementResource.create(persistentNamespaceContext,
+                namespaces,
+                redirectRetryTargeter);
+
         createDirectoryForLeaderForEachClientUseCase(NAMESPACE_1);
         createDirectoryInRootDataDirectory(NAMESPACE_2);
     }
 
     @Test
-    public void doesNotLoadLeaderPaxosAsNamespace() {
-        Set<String> namespaces = diskNamespaceLoader.getAllPersistedNamespaces().stream().map(Client::value).collect(
-                Collectors.toSet());
+    public void doesNotLoadLeaderPaxosAsNamespace() throws ExecutionException, InterruptedException {
+        Set<String> namespaces = timeLockManagementResource.getNamespaces(AUTH_HEADER).get();
```

Comment:
Do we need to create the leader paxos directory somewhere before this?

Reply:
we do create one is the setup - `createDirectoryForLeaderForEachClientUseCase`, it is of `LeaderForEachClient` paxos use case pattern but I think that's sufficient for this test.

Reply:
I see. Yep, that makes sense